src/sdl2/hint.rs

```diff
@@ -74,6 +75,47 @@ pub fn get_video_minimize_on_focus_loss() -> bool {
     )
 }
 
+/// A hint that overrides the user's locale settings.
+///
+/// [Official SDL documentation](https://wiki.libsdl.org/SDL2/SDL_HINT_PREFERRED_LOCALES)
+///
+/// # Default
+/// This is disabled by default.
+///
+/// # Example
+///
+/// See [`crate::locale::get_preferred_locales`].
+pub fn set_preferred_locales<T: std::borrow::Borrow<Locale>>(
+    locales: impl IntoIterator<Item = T>,
+) -> bool {
+    let Ok(formatted) = format_locale_hint(locales) else {
+        return false;
+    };
+    set(PREFERRED_LOCALES, &formatted)
+}
+
+fn format_locale_hint<T: std::borrow::Borrow<Locale>>(
+    locales: impl IntoIterator<Item = T>,
+) -> Result<String, std::fmt::Error> {
+    use std::fmt::Write;
+
+    let mut iter = locales.into_iter();
+    let (reserve, _) = iter.size_hint();
+    // Assuming that most locales will be of the form "xx_yy",
+    // plus 1 char for the comma.
+    let mut formatted = String::with_capacity(reserve * 6);
+
+    if let Some(first) = iter.next() {
+        write!(formatted, "{}", first.borrow())?;
+    }
+
+    for locale in iter {
+        write!(formatted, ",{}", locale.borrow())?;
+    }
```

Comment:
When using `write!` on a string (`&mut String` specifically), no errors are produced, `Ok(())` is returned every time. Handling the error by returning `Result` is not necessary. The reason the type system doesn't prevent you from doing it is because the formatting traits, function and macros use dynamic dispatch.